src/check-reserve-update-frequency/check-reserve-update-frequency.js

```diff
@@ -0,0 +1,166 @@
+function createAlert(reserveToken, oracleAge, priceSourceAddress) {
+  return Finding.fromObject({
+    name: `Stale AAVE Price Oracle Data for ${reserveToken.symbol}`,
+    description: `Token ${reserveToken.symbol} Price Oracle Age: ${oracleAge} seconds`,
+    alertId: 'AE-AAVE-PRICE-ORACLE-STALE',
+    severity: FindingSeverity.Medium,
+    type: FindingType.Suspicious,
+    everestId: '0xa3d1fd85c0b62fa8bab6b818ffc96b5ec57602b6',
+    metadata: {
+      symbol: reserveToken.symbol,
+      tokenAddress: reserveToken.tokenAddress,
+      oracleAge,
+      priceSourceAddress,
+    },
+  });
+}
+
+async function initializeTokensContracts() {
+  // create instances of ethers contracts to call read-only methods
+  const protocolDataProviderContract = new ethers.Contract(
+    protocolDataProviderAddress, protocolDataProviderAbi, jsonRpcProvider,
+  );
+  const lendingPoolAddressProviderContract = new ethers.Contract(
+    lendingPoolAddressesProvider, lendingPoolAddressesProviderAbi, jsonRpcProvider,
+  );
+
+  // get an array of all of the reserve tokens, in the form of TokenData struct entries
+  // ref: https://docs.aave.com/developers/the-core-protocol/protocol-data-provider#getallreservestokens
+  let reserveTokenArray = await protocolDataProviderContract.getAllReservesTokens();
+
+  // remove tokens that do not have Chainlink price oracles
+  reserveTokenArray = reserveTokenArray.filter(
+    (reserveToken) => TOKENS_WITHOUT_CHAINLINK_ABI.indexOf(reserveToken.tokenAddress) === -1,
+  );
+
+  // get the AAVE price oracle address
+  const priceOracleAddress = await lendingPoolAddressProviderContract.getPriceOracle();
+  const priceOracleContractInstance = new ethers.Contract(
+    priceOracleAddress,
+    priceOracleAbi,
+    jsonRpcProvider,
+  );
+
+  // get the price source addresses
+  const priceSourceAddresses = await Promise.all(reserveTokenArray.map(
+    (reserveToken) => priceOracleContractInstance.getSourceOfAsset(reserveToken.tokenAddress),
+  ));
+
+  // create ethers contracts to run read-only methods from the Chainlink contracts
+  const priceSourceContractInstances = priceSourceAddresses.map(
+    (priceSourceAddress) => new ethers.Contract(
+      priceSourceAddress,
+      chainlinkAggregatorAbi,
+      jsonRpcProvider,
+    ),
+  );
+
+  // create an array of token / address / contract tuples that we will iterate over
+  const tokenAddressContractTuples = reserveTokenArray.map((token, index) => {
+    const address = priceSourceAddresses[index];
+    const contract = priceSourceContractInstances[index];
+    return [token, address, contract];
+  });
+
+  return tokenAddressContractTuples;
+}
+
+function provideHandleBlock(tokensAddressesContractsPromise) {
+  return async function handleBlock(blockEvent) {
+    const findings = [];
+
+    // settle the promise the first time, all subsequent times just get the resolve() value
+    const tokensAddressesContracts = await tokensAddressesContractsPromise;
```

Comment:
no error checking on this, do we care? I think no :) up to you

Reply:
I agree that if this fails, we should just let it fail.